frontend/src/components/FundsPullPreApproval/ApprovalDetails.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useContext } from "react";
+import { Approval } from "../../interfaces/approval";
+import { diemAmountToHumanFriendly } from "../../utils/amount-precision";
+import { settingsContext } from "../../contexts/app";
+
+interface RequestDetailsProps {
+  approval: Approval;
+}
+
+function ApprovalDetails({ approval }: RequestDetailsProps) {
+  const [settings] = useContext(settingsContext)!;
+
+  const dateToDisplayByApprovalStatus = () => {
+    if (approval.status === "pending") {
+      return (
+        <>
+          {"on"} <span>{new Date(approval.created_at).toLocaleString()}</span>
+        </>
+      );
+    }
+    if (approval.status === "rejected") {
+      return (
+        <div>
+          {"Rejected on"} <span>{new Date(approval.updated_at).toLocaleString()}</span>
+        </div>
+      );
+    }
+    if (approval.status === "closed") {
+      return (
+        <>
+          <div>
+            {"Approved on"} <span>{new Date(approval.approved_at).toLocaleString()}</span>
```

Comment:
As a general note, you don't need to add brackets for just strings if they're inside a react element